### PR TITLE
feat: add recognition training script

### PR DESCRIPTION
**Pull request by charlesmindee**
This PR implements a recognition training script in references.
Any feedback is welcome!

PS: loss functions and vocabulary are not defined yet

### CODE REVIEW

File: references/recognition/README.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+
+The sample training script was made to train text recognition model with doctr
```
Comment (fg-mindee) [inline references/recognition/README.md:2]: Not sure we need a README if we put that little information in it :sweat_smile:

**Review by fg-mindee — COMMENTED** [review]
Thanks for the PR! Could you add usage instructions in the README ?
And can you confirm that the script works?

**Review by fg-mindee — APPROVED** [review]
Looks good to me!

**Comment by fg-mindee** [comment]
My bad, flake8 is failing :upside_down_face:

File: references/recognition/train.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    val_summary_writer = tf.summary.create_file_writer(val_log_dir)
+
+    @tf.function
+    def train_step(x, y):
+        with tf.GradientTape() as tape:
+            if args.teacher_forcing is True:
+                train_logits = model(x, y, training=True)
+            else:
+                train_logits = model(x, training=True)
+            train_loss = "!!!LOSS FN!!!"
```
Comment (fg-mindee) [inline references/recognition/train.py:69]: To avoid flake8 failing, I suggest `train_loss = train_logits.sum()`

Comment (fg-mindee) [reply references/recognition/train.py:69]: with a comment `# FIXME`

**Review by fg-mindee — CHANGES_REQUESTED** [review]
Only need to handle flake8 and we can merge :+1:

**Review by fg-mindee — APPROVED** [review]
Thanks!
